client/src/components/carlist/Carlist.js

```diff
@@ -1,60 +1,121 @@
 import AutoSelect from "../autocomplete/Autocomplete";
 import ModelSelect from "../modelac/Modelac";
 import Button from "@mui/material/Button";
-const {getCars} = require('../../service/service')
+import axios from "axios";
 
+function Carlist() {
+    const [carsmade, setCarsmade] = useState([]);
+    const [bodytypes, setBodytypes] = useState([]);
+    const [enginetypes, setEnginetypes] = useState([]);
+    const [gearboxes, setGearboxes] = useState([]);
+    const [drivetypes,setDrivetypes] = useState([])
 
+    const [loaded, setLoaded] = useState(false);
+    useEffect(()=>{
+        const fetchAllCarbrands = async ()=>{
+            try{
+                const res = await axios.get("http://localhost:4000/made")
+                setCarsmade(res.data)
+                setLoaded(true)
+            }catch (err) {
+                console.log(err)
+            }
+        }
+        fetchAllCarbrands()
+        const fetchAllBodytypes = async ()=>{
+            try{
+                const res = await axios.get("http://localhost:4000/bodytypes")
+                setBodytypes(res.data)
+                setLoaded(true)
+            }catch (err) {
+                console.log(err)
+            }
+        }
+        fetchAllBodytypes()
+        const fetchAllEnginetypes = async ()=>{
+            try{
+                const res = await axios.get("http://localhost:4000/engines")
+                setGearboxes(res.data)
+                setLoaded(true)
+            }catch (err) {
+                console.log(err)
+            }
+        }
+        fetchAllEnginetypes()
 
-function Carlist() {
-    const [cars, setCars] = useState([]);
-    useEffect(() => {getCars().then((res)=>setCars(res.data))}, []);
-    const carsmade=[]
-    const carsgb=[]
-    const carsbt=[]
-    const carseng=[]
-    const carstrans=[]
-    const carsmodel=[]
-    cars.map(car=>{
-        carsmade.push(car.made)
-        carsgb.push(car.gearbox)
-        carsbt.push(car.bodytpe)
-        carseng.push(car.enginetpe)
-        carstrans.push(car.transmission)
-    })
-    const ucarsmade = new Set(carsmade)
-    const ucarsgb = new Set(carsgb)
-    const ucarsbt = new Set(carsbt)
-    const ucarseng = new Set(carseng)
-    const ucarstrans = new Set(carstrans)
-    const ucarsmodel = new Set(carsmodel)
+        const fetchAllGearboxes = async ()=>{
+            try{
+                const res = await axios.get("http://localhost:4000/gearboxes")
+                setEnginetypes(res.data)
+                setLoaded(true)
+            }catch (err) {
+                console.log(err)
+            }
+        }
+        fetchAllGearboxes()
+
+        const fetchAllTransmissions = async ()=>{
+            try{
+                const res = await axios.get("http://localhost:4000/transmissions")
+                setDrivetypes(res.data)
+                setLoaded(true)
+            }catch (err) {
+                console.log(err)
+            }
+        }
```

Comment:
U call 5 APIs separately, if one of them crush other will calling, create one query to db that will return all data, for example. Current APIs on the backend doesn't remove. And extract all axios APIs on FE to separate service